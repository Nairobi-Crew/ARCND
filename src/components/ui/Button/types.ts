import { ButtonHTMLAttributes, FC } from 'react';

export type OwnButtonProps = {
  className?: string
} & ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = FC<OwnButtonProps>;
