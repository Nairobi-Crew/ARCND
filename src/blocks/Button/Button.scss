@import '../../common/styles/common.scss';

.button {
  font-size: 14px;
  line-height: 16px;
  border: none;
  outline: none;
  cursor: pointer;

  &_type_rounded {
    padding: 16px;
    border-radius: 24px;
  }

  &_type_rect {
    padding: 10px 16px;
    border-radius: 4px;
  }

  &_color_aqua {
    background-color: $aqua;
  }

  &_color_violet {
    background-color: $violet;
  }
}

