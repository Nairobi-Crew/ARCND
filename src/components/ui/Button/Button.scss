@import 'src/common/common';

.button {
  font-size: $f-s;
  font-family: $ff;
  line-height: 16px;
  border: none;
  outline: none;
  cursor: pointer;
  @include font-color;
  @include background-color;
  @include shadow-small;
  margin: 0.5em;
  transition: all linear 0.3s;

  &:active {
    transition: all linear 0.3s;
    transform: scale(0.7);
  }

  &_type_rounded {
    padding: 16px;
    border-radius: 16px;
  }

  &_type_rect {
    padding: 10px 16px;
    border-radius: 2px;
  }

  &_secondary {
    background-color: $secondary-btn;
  }

  &_primary {
    background-color: $primary-btn;
  }
}
