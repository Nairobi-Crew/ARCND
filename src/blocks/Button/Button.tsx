import React from 'react'
import '../../common/styles/common.scss'
import './Button.scss'

interface IButtonProps {
  type?:'button'|'reset'|'submit';
  onClick?: (e: React.MouseEvent<Element, MouseEvent>) => void;
  children?:any;
}

export default function Button({type, onClick, children}:IButtonProps) {
  return (
    <button className='button' type={type} onClick={onClick}>
      {children}
    </button>
  )
}
