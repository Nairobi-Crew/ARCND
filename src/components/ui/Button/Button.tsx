import React from 'react';
import { ButtonProps } from 'UI/Button/types';
import './Button.scss';

const Button: ButtonProps = ({ children, ...restProps }) => {
  const className = '.button';
  return (
    <button type="button" className={className} {...restProps}>{children}</button>
  );
};

export default Button;
