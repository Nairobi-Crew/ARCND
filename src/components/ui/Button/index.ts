import Button from 'UI/Button/Button';

export default Button;
