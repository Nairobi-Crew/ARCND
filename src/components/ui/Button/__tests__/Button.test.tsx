/* eslint-disable no-undef */
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import React from 'react';
import Button from 'UI/Button/index';

describe('Render Button', () => {
  it('Render correct', () => {
    const Component = () => (<Button buttonType="round" onClick={(e) => { e.preventDefault(); }}>Test</Button>);
    const buttonComponent = renderer.create(<Component />).toJSON();
    expect(buttonComponent).toMatchSnapshot();
  });

  it('buttonType="rect"', () => {
    const Component = () => (<Button buttonType="rect">Test rect button</Button>);
    const wrapper = mount(<Component />);
    expect(wrapper.find('.button_type_rect')).toHaveLength(1);
    expect(wrapper.find('.button_type_rounded')).toHaveLength(0);
  });
  it('buttonType="round"', () => {
    const Component = () => (<Button buttonType="round">Test rect button</Button>);
    const wrapper = mount(<Component />);
    expect(wrapper.find('.button_type_rect')).toHaveLength(0);
    expect(wrapper.find('.button_type_rounded')).toHaveLength(1);
  });
});
