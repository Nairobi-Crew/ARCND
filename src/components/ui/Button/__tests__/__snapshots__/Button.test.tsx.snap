// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Render correct 1`] = `
<button
  className="button button_type_rounded"
  onClick={[Function]}
>
  Test
</button>
`;
